### PR TITLE
Update TextFormatEncodingVisitor.swift

### PR DESCRIPTION
**Pull request by pruthvikar**
Fixing issue with exclusive memory access raised in Xcode 9.0-beta 3

### CODE REVIEW

**Comment by pruthvikar** [comment]
Note: This is the issue mentioned in this proposal [0176-enforce-exclusive-access-to-memory](https://github.com/apple/swift-evolution/blob/master/proposals/0176-enforce-exclusive-access-to-memory.md)

File: Sources/SwiftProtobuf/TextFormatEncodingVisitor.swift [hunk 1]
```diff
@@ -465,9 +465,11 @@ internal struct TextFormatEncodingVisitor: Visitor {
   }
 
   mutating func visitPackedEnumField<E: Enum>(value: [E], fieldNumber: Int) throws {
+    var tempEncoder = encoder
     try _visitPacked(value: value, fieldNumber: fieldNumber) { (v: E) in
-      encoder.putEnumValue(value: v)
+      tempEncoder.putEnumValue(value: v)
     }
+    encoder = tempEncoder
```
Comment (tbkka) [inline Sources/SwiftProtobuf/TextFormatEncodingVisitor.swift]: This doesn't look right to me.  Does the test suite still pass with this change?

**Comment by tbkka** [comment]
I believe that #612 (which I just merged) fixes this.

**Comment by zwaldowski** [comment]
Checking out #612 to see if that catches it… in our local copy we changed the signature of `_visitPacked` to `(value: [T], fieldNumber: Int, encode: (inout TextFormatEncoder, T) -> ()) throws` and that took care of it.

**Comment by zwaldowski** [comment]
Yep, #612 caught it. I think this can be closed. @tbkka are we expecting a tag soon, or still powering through to 1.0?

**Comment by tbkka** [comment]
@zwaldowski:  Thanks for verifying.  Closing.

(I'll push a new tag soon.)

**Pull request closed without merging**